### PR TITLE
log events when user disables the learning diary #214

### LINKED ISSUE #214: log events when user disables the learning diary
Es soll protokolliert werden, wann der nutzer das Lerntagebuch (LTB) ein und ausschaltet. 

Das ist unter anderem notwendig für die Berechnung der Lerndauer einezelner Lerneinheiten.

### PR DESCRIPTION
**Pull request by palfner-sse**
closes #214

### CODE REVIEW

File: libs/data-access/api/src/lib/trpc/routers/settings.router.ts [hunk 1]
```diff
@@ -18,6 +18,22 @@ export const settingsRouter = t.router({
...
+				await database.eventLog.create({
+					data: {
+						type: "LTB_TOGGLE",
+						payload: { enabled: input.settings.hasLearningDiary },
+						username: ctx.user.name
+					}
+				});
+			}
+
 			return await database.studentSettings.upsert({
```
Comment (spark-sse) [inline libs/data-access/api/src/lib/trpc/routers/settings.router.ts:37]: beide statements sollten zusammen in eine transaction. ansonsten wird das LTB_TOGGLE eingetragen, selbst dann, wenn danach das upsert fehlschlägt.

File: libs/data-access/api/src/lib/trpc/routers/settings.router.ts [hunk 2]
```diff
@@ -18,6 +18,22 @@ export const settingsRouter = t.router({
...
+					username: ctx.user.name
+				}
+			});
+
+			if (currentSettings?.hasLearningDiary !== input.settings.hasLearningDiary) {
+				await database.eventLog.create({
+					data: {
+						type: "LTB_TOGGLE",
+						payload: { enabled: input.settings.hasLearningDiary },
+						username: ctx.user.name
```
Comment (spark-sse) [inline libs/data-access/api/src/lib/trpc/routers/settings.router.ts:32]: etwas kontrovers, allerdings würde ich hier, die userId oder username auf die ressourceId setzen. 

`username` ist der "invoker" (evtl wird auch mal der name geändert). Die resourceId ist die geänderte ressource. 

Wenn wir später protokollieren, falls ein admin für einen anderen nutzer einstellungen ändert, wird das so auch erfasst. 

sind alles edge cases, allerdings mehr consistent

**Review by spark-sse — CHANGES_REQUESTED** [review]
bitte test für schreiben, ob informationen in der datenbank sind.

File: libs/data-access/api/src/lib/trpc/routers/settings.router.ts (hunk 2 above)
Comment (palfner-sse) [reply libs/data-access/api/src/lib/trpc/routers/settings.router.ts:32]: Wenn das in einem passiert, wird aber auch ein Toggle Event aufgerufen, wenn man Statistiken aktiviert. Dann wird der aktuelle Status des LTB (ausgeschaltet) noch einmal als Event in den Log geschrieben. Wenn das kein Problem ist, kann ich das in eins tun. Sonst nicht.

File: libs/data-access/api/src/lib/trpc/routers/settings.router.ts (hunk 1 above)
Comment (palfner-sse) [reply libs/data-access/api/src/lib/trpc/routers/settings.router.ts:37]: Wenn das in einem passiert, wird aber auch ein Toggle Event aufgerufen, wenn man Statistiken aktiviert. Dann wird der aktuelle Status des LTB (ausgeschaltet) noch einmal als Event in den Log geschrieben. Wenn das kein Problem ist, kann ich das in eins tun. Sonst nicht.

**Review by spark-sse — APPROVED** [review]
